sdk/core/Azure.Core.Experimental/src/Primitives/BinaryData.cs

```diff
@@ -13,58 +13,80 @@
 namespace Azure.Core
 {
     /// <summary>
-    /// Abstraction for a payload of binary data.
+    /// A lightweight abstraction for a payload of UTF-8 encoded bytes. This type integrates with <see cref="ObjectSerializer"/>
+    /// to allow for serializing and deserializing payloads.
+    ///
+    /// The ownership model of the underlying bytes varies depending on how the instance is constructed:
+    ///
+    /// If created using the static factory method, <see cref="FromMemory(ReadOnlyMemory{byte})"/>,
+    /// the passed in bytes will be wrapped, rather than copied. This is useful in scenarios where performance
+    /// is critical and/or ownership of the bytes is controlled completely by the consumer, thereby allowing the
+    /// enforcement of whatever ownership model is needed.
+    ///
+    /// If created using the <see cref="BinaryData(ReadOnlySpan{byte})"/> constructor, <see cref="BinaryData"/> will
+    /// maintain its own copy of the underlying bytes. This usage is geared more towards scenarios where the ownership
+    /// of the bytes might be ambiguous to users of the consuming code. By making a copy of the bytes, the payload is
+    /// guaranteed to be immutable.
+    ///
+    /// For all other constructors and static factory methods, BinaryData will assume ownership of the underlying bytes.
     /// </summary>
     public readonly struct BinaryData
     {
         private const int CopyToBufferSize = 81920;
 
+        private static readonly UTF8Encoding s_encoding = new UTF8Encoding(false);
+
         /// <summary>
         /// The backing store for the <see cref="BinaryData"/> instance.
         /// </summary>
-        internal ReadOnlyMemory<byte> Data { get; }
+        public ReadOnlyMemory<byte> Bytes { get; }
 
         /// <summary>
-        /// Creates a binary data instance from bytes.
+        /// Creates a binary data instance by making a copy
+        /// of the passed in bytes.
         /// </summary>
         /// <param name="data">Byte data.</param>
-        public BinaryData(ReadOnlyMemory<byte> data)
+        public BinaryData(ReadOnlySpan<byte> data)
         {
-            Data = data;
+            Bytes = data.ToArray();
         }
 
         /// <summary>
-        /// Creates a binary data instance from a string
-        /// using UTF-8 encoding.
+        /// Creates a binary data instance by wrapping the
+        /// passed in bytes.
+        /// </summary>
+        /// <param name="data">Byte data.</param>
+        private BinaryData(ReadOnlyMemory<byte> data)
+        {
+            Bytes = data;
+        }
+        /// <summary>
+        /// Creates a binary data instance from a string by converting
+        /// the string to bytes using UTF-8 encoding.
```

Comment:
Perhaps a `<remarks>` section that mentions that a BOM is not included as part of the conversion process?